quark/mdk-2.0.q

```diff
@@ -292,38 +313,116 @@ namespace mdk {
             }
         }
 
+        Object get(String property) {
+            return _context.properties[property];
+        }
+
+        void set(String property, Object value) {
+            _context.properties[property] = value;
+        }
+
+        bool has(String property) {
+            return _context.properties.contains(property);
+        }
+
+        void route(String service, String version, String target, String targetVersion) {
+            Map<String,List<Map<String,String>>> routes;
```

Comment:
I'm not following what (a) is referring to...

Regarding (b), if we use that structure then we would need to iterate over the keys when we look things up, and that would result in non deterministic behavior because the underlying maps won't preserve the order once the structure is deserialized.